src/ng/directive/ngModel.js

```diff
@@ -1095,141 +1251,97 @@ var DEFAULT_REGEXP = /(\s+|^)default(\s+|$)/;
-        <form name="userForm">
-          Name:
-          <input type="text" name="userName"
-                 ng-model="user.name"
-                 ng-model-options="{ debounce: 1000 }" />
-          <button ng-click="userForm.userName.$rollbackViewValue(); user.name=''">Clear</button><br />
-        </form>
-        <pre>user.name = <span ng-bind="user.name"></span></pre>
-      </div>
-    </file>
-    <file name="app.js">
-      angular.module('optionsExample', [])
-        .controller('ExampleController', ['$scope', function($scope) {
-          $scope.user = { name: 'say' };
-        }]);
-    </file>
-  </example>
+  var DEFAULT_REGEXP = /(\s+|^)default(\s+|$)/;
+  var self = this;
 
-  This one shows how to bind to getter/setters:
+  extend(this, parent || {}, options);
 
-  <example name="ngModelOptions-directive-getter-setter" module="getterSetterExample">
-    <file name="index.html">
-      <div ng-controller="ExampleController">
-        <form name="userForm">
-          Name:
-          <input type="text" name="userName"
-                 ng-model="user.name"
-                 ng-model-options="{ getterSetter: true }" />
-        </form>
-        <pre>user.name = <span ng-bind="user.name()"></span></pre>
-      </div>
-    </file>
-    <file name="app.js">
-      angular.module('getterSetterExample', [])
-        .controller('ExampleController', ['$scope', function($scope) {
-          var _name = 'Brian';
-          $scope.user = {
-            name: function(newName) {
-              return angular.isDefined(newName) ? (_name = newName) : _name;
-            }
-          };
-        }]);
-    </file>
-  </example>
- */
-var ngModelOptionsDirective = function() {
-  return {
-    restrict: 'A',
-    controller: ['$scope', '$attrs', function($scope, $attrs) {
-      var that = this;
-      this.$options = copy($scope.$eval($attrs.ngModelOptions));
-      // Allow adding/overriding bound events
-      if (this.$options.updateOn !== undefined) {
-        this.$options.updateOnDefault = false;
-        // extract "default" pseudo-event from list of events that can trigger a model update
-        this.$options.updateOn = trim(this.$options.updateOn.replace(DEFAULT_REGEXP, function() {
-          that.$options.updateOnDefault = true;
-          return ' ';
-        }));
-      } else {
-        this.$options.updateOnDefault = true;
-      }
-    }]
-  };
-};
+  // do extra processing on the options
 
+  // updateOn and updateOnDefault
+  if (isDefined(options.updateOn)) {
+    this.updateOnDefault = false;
+    // extract "default" pseudo-event from list of events that can trigger a model update
+    this.updateOn = trim(options.updateOn.replace(DEFAULT_REGEXP, function() {
+      self.updateOnDefault = true;
+      return ' ';
+    }));
+  } else if (parent) {
+    this.updateOn = parent.updateOn;
+    this.updateOnDefault = parent.updateOnDefault;
+  } else {
+    this.updateOnDefault = true;
```

Comment:
I think there's a missing test for this case (assuming I understand the case correctly):
```js
it('should update on default in case default options do not include updateOn', function () {
  module(function ($modelOptionsProvider) {
    delete $modelOptionsProvider.defaultOptions.updateOn;
  });
  inject(function ($rootScope) {
    var inputElm = helper.compileInput('<input type="text" ng-model="name" name="alias" />');
    var inputOptions = $rootScope.form.alias.$options;
    expect(inputOptions.updateOnDefault).toEqual(true);
  });
});
```